GoogleDataTransportCCTSupport/GDTCCTTests/Integration/GDTFLLIntegrationTest.m

```diff
@@ -104,74 +108,95 @@ - (void)testSendingDataToFLL {
     return;
   }
 
-  NSUInteger lengthOfTestToRunInSeconds = 10;
-  dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
-  dispatch_source_t timer = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, queue);
-  dispatch_source_set_timer(timer, DISPATCH_TIME_NOW, 1 * NSEC_PER_SEC, 0.1 * NSEC_PER_SEC);
-  dispatch_source_set_event_handler(timer, ^{
-    static int numberOfTimesCalled = 0;
-    numberOfTimesCalled++;
-    if (numberOfTimesCalled < lengthOfTestToRunInSeconds) {
-      [self generateEvent];
-    } else {
-      dispatch_source_cancel(timer);
+  // Send a number of events across multiple queues in order to ensure the threading is working as
+  // expected.
+  dispatch_queue_t queue1 = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+  dispatch_queue_t queue2 = dispatch_queue_create("com.gdtfll.test", DISPATCH_QUEUE_SERIAL);
+  for (int i = 0; i < 12; i++) {
+    int result = i % 3;
+    if (result == 0) {
```

Comment:
i think it's fine, but for future reference, modulo math is not a uniform distribution.

Reply:
SG - That's only the case for using it combined with random number generation, right? If I'm looping through 0 to 11, modulo should work as expected.

Reply:
LGTM, [ignore me](https://media.tenor.com/images/8401ff8f84969376306da37a4d276e07/tenor.gif), I'm dumb

Reply:
😄  no worries!